vm/stardust/vm_stardust.go

```diff
@@ -266,22 +265,21 @@ func accountBlockIssuerSTVF(input *vm.ChainOutputWithCreationTime, next *iotago.
 	}
 
 	if vmParams.WorkingSet.Commitment == nil {
-		return ierrors.Wrap(iotago.ErrInvalidBlockIssuerTransition, "no commitment provided")
+		return ierrors.Wrap(iotago.ErrInvalidBlockIssuerTransition, "block issuer feature validation requires a commitment input")
 	}
 
-	txSlotIndex := vmParams.WorkingSet.Commitment.Index
-	if currentBIFeat.ExpirySlot >= txSlotIndex {
+	commitmentInputIndex := vmParams.WorkingSet.Commitment.Index
+	if currentBlockIssuerFeat.ExpirySlot >= commitmentInputIndex {
 		// if the block issuer feature has not expired, it can not be removed.
-		if nextBIFeat == nil {
+		if nextBlockIssuerFeat == nil {
 			return ierrors.Wrap(iotago.ErrInvalidBlockIssuerTransition, "cannot remove block issuer feature until it expires")
 		}
-		if nextBIFeat.ExpirySlot != 0 && nextBIFeat.ExpirySlot != currentBIFeat.ExpirySlot && nextBIFeat.ExpirySlot < txSlotIndex+vmParams.API.ProtocolParameters().EvictionAge() {
+		if nextBlockIssuerFeat.ExpirySlot != currentBlockIssuerFeat.ExpirySlot && nextBlockIssuerFeat.ExpirySlot < vmParams.PastBoundedSlotIndex(commitmentInputIndex) {
 			return ierrors.Wrap(iotago.ErrInvalidBlockIssuerTransition, "block issuer feature expiry set too soon")
 		}
-
-	} else if nextBIFeat != nil {
+	} else if nextBlockIssuerFeat != nil {
 		// if the block issuer feature has expired, it must either be removed or expiry extended.
-		if nextBIFeat.ExpirySlot != 0 && nextBIFeat.ExpirySlot < txSlotIndex+vmParams.API.ProtocolParameters().EvictionAge() {
+		if nextBlockIssuerFeat.ExpirySlot < vmParams.PastBoundedSlotIndex(commitmentInputIndex) {
```

Comment:
Would it maybe make sense to calculate the `vmParams.PastBoundedSlotIndex(commitmentInputIndex` only once at the beginning of this function? It's used several times.

Maybe also the `vmParams.PastBoundedSlotIndex(commitmentInputIndex)+vmParams.API.ProtocolParameters().MaxCommittableAge())` ?

Reply:
Isn't it something like `maxCommitablePastBoundedSlot` :sweat_smile: ?

Reply:
How about minLockedSlotIndex as this value is the minimum future slot that we require an output to be locked until. So we will have <minLockedSlotIndex := pastBoundedSlotIndex + vmParams.API.ProtocolParameters().MaxCommittableAge()>